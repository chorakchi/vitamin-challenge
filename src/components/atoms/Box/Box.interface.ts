import { HTMLAttributes } from 'react';

export type shadow =
  | "01dp"
  | "02dp"
  | "03dp"
  | "04dp"
  | "06dp"
  | "08dp"
  | "09dp"
  | "12dp"
  | "16dp"
  | "24dp"

export interface BoxProps extends HTMLAttributes<HTMLElement> {

  shadow?: shadow;
  /**
   * If true, the text will have a bottom margin.
   */
  gutter?: boolean;
}