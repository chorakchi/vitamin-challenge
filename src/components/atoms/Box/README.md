The Box component serves as a wrapper component for most of the CSS utility needs.

The Box component wraps your component. It creates a new DOM element, a div that by default can be changed with the component prop.