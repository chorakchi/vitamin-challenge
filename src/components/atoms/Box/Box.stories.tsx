import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Box } from "./Box";
// @ts-ignore
import readme from "./README.md";

export default {
  title: "Atoms/Box",
  component: Box,
  parameters: {
    backgrounds: {
      default: "light",
    },
    docs: {
      description: {
        component: readme,
      },
    },
  },
} as ComponentMeta<typeof Box>;

const Template: ComponentStory<typeof Box> = (args) => <Box {...args} />;

export const Box_ = Template.bind({});
Box_.args = {
  children:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
  gutter: true,
};
