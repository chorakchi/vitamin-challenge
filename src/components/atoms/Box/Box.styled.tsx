import styled, { css } from "styled-components";
import { BoxProps } from "./Box.interface";

export const BoxStyled = styled.div<BoxProps>`
  background-color: white;
  width: fit-content;
  padding: ${({gutter}) => (gutter ? "30px" : 0)};
  margin: 60px;
  border-radius: 8px;
  ${({ shadow }) => {
    switch (shadow) {
      case "01dp":
        return css`
          box-shadow: 0 0.063rem 0.188rem rgba(69, 151, 255, 0.29),
            0 0.125rem 0.125rem rgba(69, 151, 255, 0.12),
            0 0 0.125rem rgba(69, 151, 255, 0.14);
        `;
      case "02dp":
        return css`
          box-shadow: 0 0.063rem 0.313rem rgba(69, 151, 255, 0.29),
            0 0.188rem 0.25rem rgba(69, 151, 255, 0.12),
            0 0.125rem 0.25rem rgba(69, 151, 255, 0.14);
        `;
      case "03dp":
        return css`
          box-shadow: 0 0.063rem 0.5rem rgba(69, 151, 255, 0.29),
            0 0.188rem 0.25rem rgba(69, 151, 255, 0.12),
            0 0.188rem 0.188rem rgba(69, 151, 255, 0.14);
        `;
      case "04dp":
        return css`
          box-shadow: 0 0.063rem 0.625rem rgba(69, 151, 255, 0.29),
            0 0.25rem 0.313rem rgba(69, 151, 255, 0.12),
            0 0.125rem 0.25rem rgba(69, 151, 255, 0.14);
        `;
      case "06dp":
        return css`
          box-shadow: 0 0.188rem 0.313rem rgba(69, 151, 255, 0.29),
            0 0.063rem 1.125rem rgba(69, 151, 255, 0.12),
            0 0.375rem 0.625rem rgba(69, 151, 255, 0.14);
        `;
      case "08dp":
        return css`
          box-shadow: 0 0.25rem 0.313rem rgba(69, 151, 255, 0.29),
            0 0.188rem 0.875rem 0.188rem rgba(69, 151, 255, 0.12),
            0 0.5rem 0.625rem 0.063rem rgba(69, 151, 255, 0.14);
        `;
      case "09dp":
        return css`
          box-shadow: 0 0.313rem 0.375rem rgba(69, 151, 255, 0.29),
            0 0.188rem 1rem 0.125rem rgba(69, 151, 255, 0.12),
            0 0.563rem 0.75rem 0.063rem rgba(69, 151, 255, 0.14);
        `;
      case "12dp":
        return css`
          box-shadow: 0 0.438rem 0.5rem rgba(69, 151, 255, 0.29),
            0 0.313rem 1.375rem 0.25rem rgba(69, 151, 255, 0.12),
            0 0.75rem 1.063rem 0.125rem rgba(69, 151, 255, 0.14);
        `;
      case "16dp":
        return css`
          box-shadow: 0 0.5rem 0.625rem rgba(69, 151, 255, 0.29),
            0 0.375rem 1.875rem 0.313rem rgba(69, 151, 255, 0.12),
            0 1rem 1.5rem 0.125rem rgba(69, 151, 255, 0.14);
        `;
      case "24dp":
        return css`
          box-shadow: 0 0.688rem 0.938rem rgba(69, 151, 255, 0.29),
            0 0.563rem 2.875rem 0.5rem rgba(69, 151, 255, 0.12),
            0 1.5rem 2.375rem 0.188rem rgba(69, 151, 255, 0.14);
        `;
      default:
        return css`
          filter: drop-shadow(0px 4px 117px rgba(69, 151, 255, 0.29));
        `;
    }
  }}
`;
