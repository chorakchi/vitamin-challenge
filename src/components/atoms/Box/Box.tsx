import { BoxStyled } from "./Box.styled";
import { BoxProps } from "./Box.interface";
export const Box: React.FC<BoxProps> = ({ ...props }: BoxProps) => {
  return <BoxStyled {...props} />;
};
