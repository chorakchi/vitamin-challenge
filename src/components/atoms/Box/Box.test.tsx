import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Box } from "./Box";

describe("Box", () => {
  it("should render the Box", () => {
    const subject = shallow(<Box>Box</Box>);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Box>Box</Box>).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
