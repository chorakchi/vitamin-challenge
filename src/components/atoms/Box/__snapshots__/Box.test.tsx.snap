// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render correctly and match to the Snapshot 1`] = `
<div
  className="Boxstyled__BoxStyled-sc-12nhtob-0 bOPHwF"
>
  Box
</div>
`;
